src/controllers/EthereumBlockchainController.ts

```diff
@@ -387,6 +437,32 @@ export class EthereumBlockchainController extends AbstractBlockchainController {
 		throw new Error('Method not implemented.');
 	}
 
+	async getTokenAttachments(msg: IEVMMessage): Promise<YlideTokenAttachment> {
+		const decodedMsgId = decodeEvmMsgId(msg.msgId);
+		const mailer = this.mailers.find(m => m.link.id === decodedMsgId.contractId);
+		if (!mailer) {
+			throw new Error('This message does not belongs to this blockchain controller');
+		}
+		if (mailer.wrapper instanceof EthereumMailerV9Wrapper) {
+			const tokenAttachmentAddress = msg.$$meta.tokenAttachment;
+			if (!tokenAttachmentAddress || tokenAttachmentAddress === ethers.constants.AddressZero) {
+				throw new Error('Message has no token attachment');
```

Comment:
Here is better to return `null` or empty array. If message has no attachments - it is not an error, it's normal behaviour